src/deploy/functions/params.ts

```diff
@@ -267,83 +396,271 @@ export async function resolveParams(
+      prompt = `Select a value for ${param.label || param.name}:`;
+      if (param.description) {
+        prompt += ` \n(${param.description})`;
+      }
+      return promptResourceString(prompt, param.input, projectId, resolvedDefault);
+    case "select":
+      prompt = `Select a value for ${param.label || param.name}:`;
+      if (param.description) {
+        prompt += ` \n(${param.description})`;
+      }
+      prompt += "\nSelect an option with the arrow keys, and use Enter to confirm your choice. ";
+      return promptSelect<string>(prompt, param.input, resolvedDefault, (res: string) => res);
+    case "text":
+    default:
+      prompt = `Enter a string value for ${param.label || param.name}:`;
+      if (param.description) {
+        prompt += ` \n(${param.description})`;
+      }
+      return promptText<string>(prompt, param.input, resolvedDefault, (res: string) => res);
   }
 }
 
 async function promptIntParam(param: IntParam, resolvedDefault?: number): Promise<number> {
   if (!param.input) {
-    const defaultToText: TextInput<number> = { type: "text", text: {} };
+    const defaultToText: TextInput<number> = { type: "text" };
     param.input = defaultToText;
   }
+  let prompt: string;
 
   switch (param.input.type) {
     case "select":
-      throw new FirebaseError(
-        "Build specified int parameter " + param.name + " with unsupported input type 'select'"
-      );
+      prompt = `Select a value for ${param.label || param.name}:`;
+      if (param.description) {
+        prompt += ` \n(${param.description})`;
+      }
+      prompt += "\nSelect an option with the arrow keys, and use Enter to confirm your choice. ";
+      return promptSelect(prompt, param.input, resolvedDefault, (res: string) => +res);
     case "text":
     default:
-      let prompt = `Enter a value for ${param.label || param.name}:`;
+      prompt = `Enter an integer value for ${param.label || param.name}:`;
       if (param.description) {
         prompt += ` \n(${param.description})`;
       }
-      let res: number;
-      while (true) {
-        res = await promptOnce({
-          name: param.name,
-          type: "number",
-          default: resolvedDefault,
-          message: prompt,
-        });
-        if (Number.isInteger(res)) {
-          return res;
+      return promptText<number>(prompt, param.input, resolvedDefault, (res: string) => {
+        if (isNaN(+res)) {
+          return { message: `"${res}" could not be converted to a number.` };
+        }
+        if (res.includes(".")) {
+          return { message: `${res} is not an integer value.` };
         }
-        logger.error(`${param.label || param.name} must be an integer; retrying...`);
+        return +res;
+      });
+  }
+}
+
+async function promptResourceString(
+  prompt: string,
+  input: ResourceInput,
+  projectId: string,
+  resolvedDefault?: string
+): Promise<string> {
+  const notFound = new FirebaseError(`No instances of ${input.resource.type} found.`);
```

Comment:
nit* this error is only used once - i don't think a var is necessary.

Reply:
Presumably, it will be, once we implement more resource types than just bucket.

Reply:
ah ha this is where I slip out the "YAGNI" card, but I think that's not productive comment on a month old PR. sgtm.